appsmithorg/appsmith
# fix: 10379 disable html parsing for long texts

## Description

Issue: Html parsing of long continuous texts in Text Widget causes the app to become unresponsive.
Fix: Disable html parsing for texts with no line breaks and total length > 5000.

Fixes #10379

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

1. Add a file picker widget and text widget to a page.
2. Link the text widget to render the data from filepicker (filepicker1.files[0].data)
3. set filetype to Base 64 in file picker.
4. Select the following file.
5. Text widget should render the text without the app becoming non-responsive.
[largebase64.txt](https://github.com/appsmithorg/appsmith/files/8496345/largebase64.txt)

## Checklist:




## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>


    // Code coverage diff between base branch:release and head branch: fix-10379 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.44 **(0.03)** | 37.87 **(0.02)** | 36.03 **(0.01)** | 56.66 **(0.03)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :red_circle: | app/client/src/widgets/TextWidget/widget/index.tsx | 100 **(0)** | 63.64 **(-11.36)** | 100 **(0)** | 100 **(0)**
 :green_circle: | app/client/src/workers/helpers.ts | 83.87 **(41.93)** | 43.75 **(31.25)** | 100 **(50)** | 88 **(52)**</details>
